File: 69-sqrtx/69-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        
        long long start= 0 , end = x , ans , mid;
        
        while( start <=end){
            mid = (start+end)/2;
            if (mid*mid==x) return mid;
            
            else if(mid*mid > x){
               if( (mid-1)*(mid-1) <x ){
                   return mid-1;
               }
               end = mid-1;       
            }  
            else start = mid+1;
        }
        return ans; 
    }
};

File: 69-sqrtx/69-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        
        if x==0 or x==1:
            return x
        for n in range(1, x):
            
             if n*n == x:
                return n 
             if (n+1)*(n+1) >x:
                    return n 